File: gr00t/Isaac-GR00T/examples/H20/robots/sdk_package/example_py/h10-w/status/get_move_message.py
import humanoid_sdk_py
from humanoid_sdk_py import h10w


h10w_status = h10w.H10wStatus()
print("=== 获取机器人当前运动状态信息 ===")

ret, move_msg = h10w_status.getMoveMessage()
if ret != 0 or move_msg is None:
    print("获取运动信息失败，任务终止")
    exit(-1)

print("状态：", move_msg.state)
print("位置：", move_msg.position)
print("速度：", move_msg.velocity)
print("机器人当前位姿：")
for tcp_pose in move_msg.tcp_pose:
    print(tcp_pose.type, tcp_pose.pose)
print("当前执行任务：", move_msg.task_id)

File: gr00t/Isaac-GR00T/examples/H20/robots/sdk_package/example/h10-w/status/get_move_message.cpp
// get_move_message.cpp

#include <iostream>
#include <iomanip>
#include "h10-w/h10w_status.h"
#include "common/common_def.h"

using namespace humanoid;
using namespace humanoid::h10w;

int main() {
  std::cout << "=== 获取机器人当前运动状态信息示例 ===" << std::endl;

  H10wStatus status;
  MoveMessageParams moveMessage;

  int result = status.getMoveMessage(moveMessage);
  if (result != 0) {
    std::cerr << "获取运动信息失败，任务终止" << std::endl;
    return -1;
  }
  std::cout << "机器人当前状态：" << static_cast<int>(moveMessage.state)
            << std::endl;
  std::cout << "机器人当前位置：" << std::endl;
  for (auto& pos : moveMessage.position) {
    std::cout << std::fixed << std::setprecision(3) << pos << " ";
  }
  std::cout << std::endl;
  std::cout << "机器人当前速度：" << std::endl;
  for (auto& vel : moveMessage.velocity) {
    std::cout << std::fixed << std::setprecision(3) << vel << " ";
  }
  std::cout << std::endl;
  std::cout << "机器人当前TCP位姿：" << std::endl;
  for (auto& pose : moveMessage.tcp_pose) {
    std::cout << "TCP " << static_cast<int>(pose.type) << " ：" << std::endl;
    for (size_t i = 0; i < 6; i++) {
      std::cout << std::fixed << std::setprecision(2) << pose.pose[i] << " ";
    }
    std::cout << std::endl;
  }
  std::cout << "机器人当前执行任务：" << moveMessage.task_id << std::endl;
  return 0;
}